src/python/pants/backend/python/rules/pex.py

```diff
@@ -67,13 +75,79 @@ def create_from_adaptors(
         return PexInterpreterConstraints(constraint_set=tuple(sorted(interpreter_constraints)))
 
 
+@dataclass(frozen=True)
+class PexRequirementConstraints:
+    """A collection of paths to constraint files and/or Requirement-style strings."""
+
+    constraint_file_paths: Tuple[str, ...] = ()
+    raw_constraints: Tuple[str, ...] = ()
+    generated_file_name: ClassVar[str] = "requirement_constraints.generated.txt"
+
+    @classmethod
+    def create_from_raw_constraints(cls, constraints: Iterable[str]) -> "PexRequirementConstraints":
+        """The constraints may be paths to constraint .txt files and/or requirement strings."""
+        constraint_file_paths = set()
+        raw_constraints = set()
+        for constraint in constraints:
+            if PurePath(constraint).suffix == ".txt":
+                constraint_file_paths.add(constraint)
+            else:
+                raw_constraints.add(constraint)
+        return PexRequirementConstraints(
+            constraint_file_paths=tuple(sorted(constraint_file_paths)),
+            raw_constraints=tuple(sorted(raw_constraints)),
+        )
+
+    @classmethod
+    def create_from_global_setup(cls, python_setup: PythonSetup) -> "PexRequirementConstraints":
```

Comment:
Maybe name `create_from_setup` ? This method isn't concerned with where the setup came from.